src/components/@molecules/SearchInput/SearchInput.tsx

```diff
@@ -338,6 +341,13 @@ const createSearchHandler =
       { lastAccessed: Date.now(), nameType, text, isValid: selectedItem.isValid },
     ])
 
+    if (nameType === 'eth' || nameType === 'box') {
+      trackEvent({
+        eventName: `search_selected_${nameType}`,
+        customProperties: { name: text },
+      })
+    }
+
     const path = getRouteForSearchItem({ address, chainId, queryClient, selectedItem })
```

Comment:
@storywithoutend
I've implemented your suggestion, but I'm encountering an issue. When a user enters a keyword and immediately presses Enter, the path always resolves to `/profile/{name}` instead of `/register/{name}`.

I believe the `ownerData` variable is `undefined` because `getCachedQueryData` isn't returning data in time.

To address this, I've added a `waitForTimeout` in `e2e/specs/stateless/registerName.spec.ts` and included a comment explaining the reason.

Please review these changes and let me know if you have any further feedback.